### ISSUE TITLE
Showing warning message if user publishes a non null-safe version

### ISSUE THREAD

**Issue by tusharojha** [dart-lang/pub#2921]
## Description

As null safety landed stable we prefer all the new packages & new versions of old packages to be null safe. It would be great if we can show a warning message if the user publishes a non-null-safe package.  

We can show a prompt for the user to confirm that they want to move forward or not.

@sigurdm @jonasfj WDYT?

**Comment by jonasfj** [comment]
While I agree that we prefer null-safety packages going forward.

It is not inherently wrong with taking some time to migrate.

I don't even think default application templates have switched to null-safety yet.

So I think we should wait. Keep in mind that if you're developing non-null-safety code you can still import null-safe packages, so you get packages from both worlds when you haven't migrated. You won't get performance benefits until all dependencies are migrated, and you have migrated your code. But unless you are doing a lot compute and hungry for more performance, waiting a bit with migrating to null-safety is rather attractive.

Besides, if your dependencies haven't migrated to null-safety yet, it's probably better to wait a bit.

-------

In other words, I think the noise ratio is too high here. Developers who care about their packages, will watch the score on pub.dev and see that they don't have a null-safety badge :D

**Comment by sigurdm** [comment]
To add on: we have warnings if your sdk-constraint lower bound is >= 2.12.0 but you have not migrated
